### PR TITLE
MRG: Fix to tests

### PR DESCRIPTION
**Pull request by jasmainak**

### CODE REVIEW

**Comment by jasmainak** [comment]
@pavanramkumar @titipata some refactoring of the scoring functions going on here to make it more readable. Something going wrong at the moment, but just a warning if either of you are working on this same code. I'll stop today. Feel free to push to my branch and merge it if you can see the light ... :)

**Comment by pavanramkumar** [comment]
@jasmainak the issue is in tests as far as i can tell by looking at the diff. `pseudo_R2` need not be positive (if training set is overfitting and test set is underfitting, it can be negative), but `deviance` must be non-negative.

**Comment by pavanramkumar** [comment]
related: `'multinomial'` is typically evaluated using a score like `accuracy` or `F1`: `deviance` and `pseudo_R2` are not common. feel free to implement this as you're fixing `score()`

**Comment by jasmainak** [comment]
@pavanramkumar I am done for now :) 

> related: 'multinomial' is typically evaluated using a score like accuracy or F1: deviance and pseudo_R2 are not common. feel free to implement this as you're fixing score()

Let's raise an issue for now? My feeling is that we should stop adding features now (or soon enough) and spend at least 15 days just doing bug fixes + documentation. I encountered many issues while trying to clean up this part of the code. Also, cross-validation doesn't seem to be working smoothly with the inbuilt `score` method. Not sure why. Let's raise an issue and fix later ...
